tracing-subscriber/src/registry/sharded.rs

```diff
@@ -715,7 +711,7 @@ mod tests {
 
         let dispatch = dispatcher::Dispatch::new(subscriber);
 
-        dispatcher::with_default(&dispatch, || {
+        dispatcher::with_default(dispatch, || {
```

Comment:
```suggestion
        dispatcher::with_default(dispatch.clone(), || {
```

Reply:
Now clippy is angry about unnecessary clones. I can
- remove the clones
- silence the clippy warning
- add `drop(dispatch)` after the `with_default` block

Going with the last but it's not very satisfying. :/